### PR TITLE
[ZEVMIDST-143] Implement MFA enrollment

### PR DESCRIPTION
**Pull request by kacpersmal**
This PR includes mfa enrollment views & logic

### CODE REVIEW

File: src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        return Task.FromResult(entry);
+    }
+
+    public async Task<bool> VerifyCode(string code)
+    {
+        var cookie = GetMfaCookie();
+        if (cookie is null)
+        {
+            _logger.LogInformation("No MFA cookie found");
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs:55]: Maybe logWarning?

File: src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+using Newtonsoft.Json;
+
+namespace DfT.ZEV.Common.MVC.Authentication.Services.Mfa;
+
+public class MfaService : IMfaService
+{
+    private readonly IOptions<GoogleCloudConfiguration> _googleOptions;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ILogger<MfaService> _logger;
+    public MfaService(IOptions<GoogleCloudConfiguration> googleOptions, IHttpContextAccessor httpContextAccessor, ILogger<MfaService> logger)
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs:17]: Add an empty line

File: src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs (hunk 1 above)
Comment (kacpersmal) [reply src/common/DfT.ZEV.Common.MVC.Authentication/Services/Mfa/MfaService.cs:55]: removed completely, GetMfaCookie automatically initializes cookie when it is not found

**Review by dae-ne — APPROVED** [review]

File: src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/MfaController.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        var notification = new Notification
+        {
+            Recipients = new List<string> { model.MobileNumber },
+            TemplateId = Guid.Parse(_googleOptions.Value.Queues.Notification.MfaVerificationTemplateId),
+            TemplateParameters = new Dictionary<string, string>
+                { { "mfa_code", code.Code }, { "service_name", _configuration.GetSection("ServiceName").Value } },
+            NotificationType = NotificationType.Sms
+        };
+        await _notificationService.SendNotificationAsync(notification);
+        _logger.LogInformation("Generated new code {Code} for user {number}", code.Code, model.MobileNumber);
```
Comment (cruddasj) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/MfaController.cs:154]: Minor but as discussed - can we wrap this in a #Debug pragma?

**Review by cruddasj — APPROVED** [review]

**Pull request merged**